cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,578 @@
+        },
+        re.compile(SECTION2.format('environment')): {
+            'short': (
+                '``[cylc][environment]`` is obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'.* handler\s*?='): {
+            'short': (
+                '``[cylc][<namespace>][events]___ handler`` commands are'
+                ' now "handlers".'
+            ),
+            'url': ''
+        },
+        re.compile(r'mail retry delays\s*?='): {
+            'short': (
+                '``[runtime][<namespace>][events]mail retry delays`` '
+                'is obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'extra log files\s*?='): {
+            'short': (
+                '``[runtime][<namespace>][events]extra log files`` '
+                'is obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'shell\s*?='): {
+            'short': (
+                '``[runtime][<namespace>]shell`` '
+                'is obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'suite definition directory\s*?='): {
+            'short': (
+                '``[runtime][<namespace>][remote]suite definition directory`` '
+                'is obsolete.'
+            ),
+            'url': 'summary.html#symlink-dirs'
+        },
+        re.compile(SECTION2.format('dependencies')): {
+            'short': '``[dependencies]`` is deprecated.',
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'graph\s*?='): {
+            'short': (
+                '``[cycle point]graph =`` is deprecated, '
+                'use ``cycle point = <graph>``'
+            ),
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(SECTION2.format('remote')): {
+            'short': (
+                '``[runtime][<namespace>][remote]host`` is deprecated, '
+                'use ``[runtime][<namespace>]platform``'
+            ),
+            'url': 'major-changes/platforms.html#platforms'
+        },
+        re.compile(r'suite state polling\s*='): {
+            'short': (
+                '``[runtime][<namespace>]suite state polling`` is deprecated, '
+                'use ``[runtime][<namespace>]workflow state polling``'
+            ),
+            'url': 'major-changes/platforms.html#platforms'
+        },
+        re.compile(SECTION3.format('job')): {
+            'short': (
+                '``[runtime][<namespace>][job]`` is deprecated, '
+                'use ``[runtime][<namespace>]platform``'
+                '\n    (the following items can be moved to '
+                '``[runtime][<namespace>]``:'
+                '\n    - ``execution retry delays``'
+                '\n    - ``submission retry delays``'
+                '\n    - ``execution polling intervals``'
+                '\n    - ``submission polling intervals``'
+                '\n    - ``execution time limit``'
```

Comment:
These extra lines aren't commented out in `-i` mode.

Maybe just this will suffice?
```suggestion
                '``[runtime][<namespace>][job]`` is deprecated'
```